rflx/cli.py

```diff
@@ -137,9 +146,28 @@ def graph(args: argparse.Namespace) -> None:
     if not directory.is_dir():
         fail(f'directory not found: "{directory}"', Subsystem.GRAPH)
 
-    model = parse(args.files)
+    model = parse(args.files, args.no_verification)
+    locations: Dict[str, Dict[str, Dict[str, Dict[str, int]]]] = {}
 
     for m in model.messages:
         message = flat_name(m.full_name)
         filename = Path(directory).joinpath(message).with_suffix(f".{args.format}")
         Graph(m).write(filename, fmt=args.format)
+
+        assert m.location
+        assert m.location.start
+        assert m.location.end
+        assert m.location.source
+
+        source = str(m.location.source.name)
+
+        if source not in locations:
+            locations[source] = {}
+        locations[source][message] = {
+            "start": {"line": m.location.start[0], "column": m.location.start[1]},
+            "end": {"line": m.location.end[0], "column": m.location.end[1]},
+        }
+
+    filename = Path(directory).joinpath("locations.json")
+    with open(filename, "w") as f:
+        json.dump(locations, f)
```

Comment:
This is only needed for the GNAT Studio integration. We should generate that only if needed (e.g. by setting an additional argument).

Reply:
I made this optional in 8cf8ba3dba47d5e77047270d2197419bd80050a2.